src/display/editor/signature.js

```diff
@@ -244,27 +244,18 @@ class SignatureEditor extends DrawingEditor {
   }
 
   /** @inheritdoc */
-  async addEditToolbar() {
-    const toolbar = await super.addEditToolbar();
-    if (!toolbar) {
-      return null;
-    }
-    if (this._uiManager.signatureManager && this.#description !== null) {
-      await toolbar.addEditSignatureButton(
-        this._uiManager.signatureManager,
-        this.#signatureUUID,
-        this.#description
-      );
-      toolbar.show();
+  get toolbarTools() {
+    if (this._uiManager.signatureManager) {
+      return [["editSignature", this._uiManager.signatureManager]];
     }
-    return toolbar;
+    return super.toolbarTools;
   }
 
   addSignature(data, heightInPage, description, uuid) {
     const { x: savedX, y: savedY } = this;
     const { outline } = (this.#signatureData = data);
     this.#isExtracted = outline instanceof ContourDrawOutline;
-    this.#description = description;
+    this.description = description;
```

Comment:
Is this correct because the class already has a public `description` getter at https://github.com/mozilla/pdf.js/blob/84c185ce1fe250865addd9fcc720d6a579f5eb65/src/display/editor/signature.js#L214 which returns the value of the private member, so aren't we essentially overwriting the getter here?

Reply:
There's a setter just after so it should be fine, right ?